lib/auth/auth_with_roles.go

```diff
@@ -1536,7 +1535,15 @@ func (a *ServerWithRoles) ListUnifiedResources(ctx context.Context, req *proto.L
 	}
 
 	unifiedResources, nextKey, err := a.authServer.UnifiedResourceCache.IterateUnifiedResources(ctx, func(resource types.ResourceWithLabels) (bool, error) {
-		if err := resourceChecker.CanAccess(resource); err != nil {
+		var err error
+		switch r := resource.(type) {
+		case types.SAMLIdPServiceProvider:
+			err = a.action(apidefaults.Namespace, types.KindSAMLIdPServiceProvider, types.VerbList)
```

Comment:
Any plans to integrate this into the resource checker?

Reply:
I'm not aware of any (for or against) but I think we should. I'm a bit unsure how the SAML stuff works outside of filtering for it here tbh